addons/mrp/models/mrp_workorder.py

```diff
@@ -195,9 +227,66 @@ def generate_wo_lines(self):
             line_values = self._generate_lines_values(move, qty_to_consume)
             self.workorder_line_ids |= self.env['mrp.workorder.line'].create(line_values)
 
-    def _assign_default_final_lot_id(self):
-        self.final_lot_id = self.env['stock.production.lot'].search([('use_next_on_work_order_id', '=', self.id)],
-                                                                    order='create_date, id', limit=1)
+    def _update_wo_lines(self):
+        """ update existing line on the workorder. It could be trigger manually
+        after a modification of qty_producing.
+        """
```

Comment:
rename this method into _apply_update_workorder_lines
apply _update_workorder_lines on self
this method should be called after updated `qty_producing` on self